qiskit/providers/models/backendconfiguration.py

```diff
@@ -122,8 +200,124 @@ def __init__(self, backend_name, backend_version, n_qubits, basis_gates,
         self.local = local
         self.simulator = simulator
         self.conditional = conditional
-        self.open_pulse = open_pulse
         self.memory = memory
         self.max_shots = max_shots
 
         super().__init__(**kwargs)
+
+    @classmethod
+    def from_dict(cls, dict_):
```

Comment:
Can you remove the customized `from_dict` methods?

I think I see the intent, but the main advantage (not breaking existing code) _should_ already be achieved by the subclassing, if done correctly (as in, if existing code instantiates a generic `BackendConfiguration` for a qasm backend in the same way as before, it would not break or actually change - specially if the change for `open_pulse` not required is reverted; and for pulse backends we have the chance of modifying them as they have not been published in a stable release yet). Removing `from_dict` would help keeping the models as just containers - I think we are better off not moving functionality into them: even if is a relatively small change from the line counts point of view, it has some implications at the arquitectural level that I'd rather avoid.

In turn, this probably means adjusting the individual backends for using the specific subclasses - can you include the changes for `basicaer` in this PR, and open an issue in the `qiskit-ibmq-provider` repo for handling those in that provider?